lib/langchain/vectorsearch/pgvector.rb

```diff
@@ -13,60 +13,70 @@ class Pgvector < Base
+      "cosine_distance" => "cosine",
+      "euclidean_distance" => "euclidean"
     }
     DEFAULT_OPERATOR = "cosine_distance"
 
-    attr_reader :operator, :quoted_table_name
+    attr_reader :db, :operator, :table_name, :namespace_column, :namespace, :documents_table
 
     # @param url [String] The URL of the PostgreSQL database
     # @param index_name [String] The name of the table to use for the index
     # @param llm [Object] The LLM client to use
     # @param api_key [String] The API key for the Vectorsearch DB (not used for PostgreSQL)
-    def initialize(url:, index_name:, llm:, api_key: nil)
-      require "pg"
+    # @param namespace_column [String] The name of the column to use for the namespace
+    # @param namespace [String] The namespace to use for the index when inserting/querying
+    def initialize(url:, index_name:, llm:, api_key: nil, namespace_column: nil, namespace: nil)
+      depends_on "sequel"
+      require "sequel"
+      depends_on "pgvector"
       require "pgvector"
 
-      @client = ::PG.connect(url)
-      registry = ::PG::BasicTypeRegistry.new.define_default_types
-      ::Pgvector::PG.register_vector(registry)
-      @client.type_map_for_results = PG::BasicTypeMapForResults.new(@client, registry: registry)
+      @db = Sequel.connect(url)
 
-      @index_name = index_name
-      @quoted_table_name = @client.quote_ident(index_name)
+      @table_name = index_name
+
+      @namespace_column = namespace_column || "namespace"
+      @namespace = namespace
       @operator = OPERATORS[DEFAULT_OPERATOR]
 
       super(llm: llm)
     end
 
+    def documents_model
+      Class.new(Sequel::Model(table_name.to_sym)) do
+        plugin :pgvector, :vectors
+      end
+    end
+
     # Add a list of texts to the index
     # @param texts [Array<String>] The texts to add to the index
     # @return [PG::Result] The response from the database
     def add_texts(texts:)
       data = texts.flat_map do |text|
-        [text, llm.embed(text: text)]
+        embedding = llm.embed(text: text)
+        {
+          :content => text,
+          :vectors => ::Pgvector.encode(embedding),
+          namespace_column.to_sym => namespace
+        }
       end
-      values = texts.length.times.map { |i| "($#{2 * i + 1}, $#{2 * i + 2})" }.join(",")
-      client.exec_params(
-        "INSERT INTO #{quoted_table_name} (content, vectors) VALUES #{values};",
-        data
-      )
+
+      documents_model.multi_insert(data, return: :primary_key)
     end
 
     # Create default schema
     # @return [PG::Result] The response from the database
     def create_default_schema
-      client.exec("CREATE EXTENSION IF NOT EXISTS vector;")
-      client.exec(
-        <<~SQL
-          CREATE TABLE IF NOT EXISTS #{quoted_table_name} (
-            id serial PRIMARY KEY,
-            content TEXT,
-            vectors VECTOR(#{default_dimension})
-          );
-        SQL
-      )
+      db.run "CREATE EXTENSION IF NOT EXISTS vector"
```

Comment:
This throws an error:
```ruby
Sequel::DatabaseError (PG::UndefinedFile: ERROR:  could not open extension control file "/Applications/Postgres.app/Contents/Versions/13/share/postgresql/extension/vector.control": No such file or directory)
```

Reply:
Does it work for you? 
``ruby
pgvector = Langchain::Vectorsearch::Pgvector.new url: url, llm: llm, index_name: "vectors"
pgvector.create_default_schema
```

Reply:
Do you have pgvector installed? Looks like it's not there for pg@13?

Reply:
Yes, works for me.